glean-core/ffi/src/lib.rs

```diff
@@ -139,5 +167,49 @@ pub extern "C" fn glean_ping_collect(glean_handle: u64, ping_type_handle: u64) -
     })
 }
 
+#[no_mangle]
+pub extern "C" fn glean_set_experiment_active(
+    glean_handle: u64,
+    experiment_id: FfiStr,
+    branch: FfiStr,
+    extra_keys: RawStringArray,
+    extra_values: RawStringArray,
+    extra_len: i32,
+) {
+    GLEAN.call_infallible(glean_handle, |glean| {
+        let extra =
+            unsafe { from_raw_string_array_and_string_array(extra_keys, extra_values, extra_len) };
+        glean.set_experiment_active(
+            experiment_id.as_str().to_string(),
```

Comment:
I'm filing an issue for us to double-check which parts of the FFI can potentially panic (this can).

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1567110